src/js/actions/tools.js

```diff
@@ -484,14 +484,18 @@ define(function (require, exports) {
     var handleToolModalStateChanged = function (event) {
         var modalState = (event.state._value === "enter"),
             changeStatePromise = this.transfer(changeModalState, modalState),
+            policyStore = this.flux.stores.policy,
+            toolsIdsThatSuspendPolicies = ["txBx", "arwT", "Rect"],
             policyPromise;
 
         // Suspend policies during type tool modal states
-        if (event.kind._value === "tool" && event.tool.ID === "txBx") {
-            if (modalState) {
+        if (event.kind._value === "tool" && toolsIdsThatSuspendPolicies.indexOf(event.tool.ID) > -1) {
+            if (modalState && !policyStore.areAllSuspended()) {
```

Comment:
note that this is actually the clean path through here.  In this case, `modalState` is maybe a bad var name because it means "entering a modal state".

If you are asking about why we'd be going in to modal state if policies might already be suspended ... that is a fair question.  It shouldn't happen, but given that PS sometimes emits those "late" modal tool states, this helps prevent overlap between this and the 'native menu' hack.